apps/shell/tash_main.c

```diff
@@ -37,7 +37,8 @@
 enum tash_input_state_e {
 	IN_VOID,
 	IN_WORD,
-	IN_QUOTE
+	IN_QUOTE,
+	IN_APOSTROPHE
```

Comment:
As I left before, we can add a new state here for `!`. I mean we can merge the functions for ! and others. Please consider this with new PR.